types/parameterized_struct_type.go

```diff
@@ -0,0 +1,80 @@
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/substrait-io/substrait-go/types/parameter_types"
+)
+
+// ParameterizedStructType is a struct having at least one parameter of type ParameterizedAbstractType
+// example: Struct<Decimal(P,S), INT8>.
+// If All arguments are concrete they are represented by StructType
+type ParameterizedStructType struct {
+	Nullability      Nullability
+	TypeVariationRef uint32
+	Type             []Type
+}
```

Comment:
since this is exactly the same definition as `StructType` maybe we just do

```go
type ParameterizedStructType struct {
        StructType
}
```

and just implement any functions that are different. or 

```go
type ParameterizedStructType StructType
```

either way that'll let us avoid having to manually create `StructType` instances in the methods.

Reply:
It was a mistake to use []Type as Type. It is []FuncDefArgType. Fixed it in updated PR to use FuncDefArgType